src/GovUk.Education.ExploreEducationStatistics.Content.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -0,0 +1,366 @@
+                    It.IsAny<PaginatedListViewModel<DataSetListViewModel>>(),
+                    It.IsAny<MemoryCacheConfiguration>(),
+                    null));
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            MockUtils.VerifyAllMocks(MemoryCacheService);
+
+            var pagedResult = response.AssertOk<PaginatedListViewModel<DataSetListViewModel>>();
+
+            AssertPaginatedViewModel(pagedResult, expectedTotalResults: 0);
+            Assert.Empty(pagedResult.Results);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public async Task PageTooSmall_ReturnsBadRequest(int page)
+        {
+            var query = new DataSetsListRequest(
+                Page: page
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            response.AssertBadRequest(expectedErrorKey: "Page",
+                expectedErrorMessage: "'Page' must be greater than or equal to '1'.");
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public async Task PageSizeTooSmall_ReturnsBadRequest(int pageSize)
+        {
+            var query = new DataSetsListRequest(
+                PageSize: pageSize
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            response.AssertBadRequest(expectedErrorKey: "PageSize",
+                expectedErrorMessage: "'Page Size' must be greater than or equal to '1'.");
+        }
+
+        [Theory]
+        [InlineData(41)]
+        public async Task PageSizeTooBig_ReturnsBadRequest(int pageSize)
+        {
+            var query = new DataSetsListRequest(
+                PageSize: pageSize
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            response.AssertBadRequest(expectedErrorKey: "PageSize",
+                expectedErrorMessage: "'Page Size' must be less than or equal to '40'.");
+        }
+
+        [Theory]
+        [InlineData("a")]
+        [InlineData("aa")]
+        public async Task SearchTermTooSmall_ReturnsBadRequest(string searchTerm)
+        {
+            var query = new DataSetsListRequest(
+                SearchTerm: searchTerm
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            MockUtils.VerifyAllMocks(MemoryCacheService);
+
+            response.AssertBadRequest(expectedErrorKey: "SearchTerm",
+                expectedErrorMessage:
+                $"The length of 'Search Term' must be at least 3 characters. You entered {searchTerm.Length} characters.");
+        }
+
+        [Fact]
+        public async Task OrderByNaturalWithoutReleaseId_ReturnsBadRequest()
```

Comment:
This one is in inner class `ValidationTests`. I didn't feel the need to provice a success case because success cases should already be covered by these tests which are in `OrderByTests`. 

```csharp
OrderByTests.OrderByNaturalAscending_Success()
OrderByTests.OrderByNaturalDescending_Success()
OrderByTests.OrderByNaturalWithoutSortOrder_DefaultsToAscending()
```

Reply:
Added success cases in ef7d583a15d99134500966305da5be3309c2eaa0, thanks.

Reply:
Ta